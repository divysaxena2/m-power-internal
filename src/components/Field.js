import React, { Component } from 'react';
import getTheme from '../../native-base-theme/components'
import material from '../../native-base-theme/variables/material';
import { StyleSheet, View } from 'react-native';
import { StyleProvider, Form, Item, Label, Input } from 'native-base';


type Props = {};
export default class Field extends Component<Props> {

    render() {
        return (
            <StyleProvider style={getTheme(material)}>
                <Form>
                    <Item floatingLabel>
                        <Label>{this.props.label}</Label>
                        <Input />
                    </Item>
                </Form>
            </StyleProvider>)
    }
}
const styles = StyleSheet.create({

});