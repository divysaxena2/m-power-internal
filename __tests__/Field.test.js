import 'react-native';
import React from 'react';


import renderer from 'react-test-renderer';
import Field from '../src/components/Field';

it('renders correctly', () => {
    const tree = renderer.create(
        <Field />
    ).toJSON();
    expect(tree).toMatchSnapshot();
});