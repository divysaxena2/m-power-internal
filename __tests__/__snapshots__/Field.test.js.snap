// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={Object {}}
>
  <View
    accessible={true}
    isTVSelectable={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "transparent",
        "borderBottomWidth": 1,
        "borderColor": "#D9D5DC",
        "borderLeftWidth": 0,
        "borderRadius": undefined,
        "borderRightWidth": 0,
        "borderTopWidth": 0,
        "borderWidth": undefined,
        "flexDirection": "row",
        "marginLeft": 15,
        "marginTop": 15,
        "opacity": 1,
      }
    }
  >
    <View
      style={
        Object {
          "left": 0,
          "opacity": 1,
          "paddingBottom": undefined,
          "paddingTop": undefined,
          "position": "absolute",
          "right": 0,
          "top": 18,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#575757",
            "fontSize": 17,
            "left": 0,
            "paddingRight": 5,
            "paddingTop": 5,
            "top": 6,
          }
        }
      >
        <Text
          style={
            Object {
              "fontSize": 17,
            }
          }
        />
      </Text>
    </View>
    <TextInput
      allowFontScaling={true}
      editable={true}
      onBlur={[Function]}
      onChangeText={[Function]}
      onFocus={[Function]}
      placeholderTextColor="#575757"
      style={
        Object {
          "color": "#000",
          "flex": 1,
          "fontSize": 17,
          "height": 50,
          "marginLeft": null,
          "paddingBottom": 7,
          "paddingLeft": null,
          "paddingRight": 5,
          "paddingTop": 3,
          "top": 10,
        }
      }
      underlineColorAndroid="rgba(0,0,0,0)"
      value=""
    />
  </View>
</View>
`;
